subprojects/jacoco/src/integTest/groovy/org/gradle/testing/jacoco/plugins/JacocoCoverageAggregationIntegrationTest.groovy

```diff
@@ -406,17 +406,12 @@ class JacocoCoverageAggregationIntegrationTest extends AbstractIntegrationSpec {
             abstract class CustomAggregation extends DefaultTask {
                 @TaskAction
                 void aggregate() {
-                    Configuration resolver = getProject().getConfigurations().create("jacocoResolver")
-                    resolver.visible = false
-                    resolver.canBeConsumed = false
-                    resolver.canBeResolved = true
-                    resolver.extendsFrom(getProject().getConfigurations().getByName("implementation"))
-                    resolver.attributes {
-                        attribute(Usage.USAGE_ATTRIBUTE, project.getObjects().named(Usage.class, Usage.JAVA_RUNTIME));
-                        attribute(Category.CATEGORY_ATTRIBUTE, project.getObjects().named(Category.class, Category.DOCUMENTATION));
-                        attribute(DocsType.DOCS_TYPE_ATTRIBUTE, project.getObjects().named(DocsType.class, "jacoco-coverage-data"));
+                    def resolver = project.configurations.resolver("jacocoResolver", JacocoCoverage) {
+                        from(project.configurations.implementation)
+                        forTestTasksNamed("test")
+                        lenient()
```

Comment:
Those are the things that the user is interested for `JacocoCoverage` type, not the individual flags that make a resolver out of configuration or the individual attributes.